motor-controller/common/uart.c

```diff
@@ -93,135 +101,171 @@ bool _uart_start_transmit_dma() {
 
     // enable DMA
     DMA1_Channel2->CCR |= DMA_CCR_EN;
-
-    return true;
 }
 
 //////////
 //  RX  //
 //////////
 
 bool uart_can_read() {
-    return (!ioq_empty(&uart_rx_queue));
+    return (!ioq_is_empty(&uart_rx_queue));
 }
 
 void uart_discard() {
-    ioq_discard(&uart_rx_queue);
+    ioq_discard_write_back(&uart_rx_queue);
```

Comment:
We seem to have lost the while loop down stream to delete more than just the last element. Should we move it here?